smithy-build/src/main/java/software/amazon/smithy/build/plugins/RunPlugin.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.build.plugins;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+import software.amazon.smithy.build.PluginContext;
+import software.amazon.smithy.build.SmithyBuildException;
+import software.amazon.smithy.build.SmithyBuildPlugin;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.node.Node;
+import software.amazon.smithy.model.node.NodeMapper;
+import software.amazon.smithy.model.shapes.ModelSerializer;
+import software.amazon.smithy.utils.IoUtils;
+import software.amazon.smithy.utils.SmithyInternalApi;
+
+/**
+ * Runs a process-based plugin.
+ */
+public final class RunPlugin implements SmithyBuildPlugin {
+    private static final Logger LOGGER = Logger.getLogger(RunPlugin.class.getName());
+    private final Path root = Paths.get(".").toAbsolutePath().normalize();
```

Comment:
We can use here instead `SmithyBuildUtils.getCurrentWorkingDirectory()`

Reply:
Can't use it here since it's package private right now. I'm ok with the duplication for now to spare needing to make it public.